src/cbc_sdk/workload/vulnerability_assessment.py

```diff
@@ -151,37 +58,36 @@ def _refresh(self):
         Returns:
             bool: True if refresh was successful, False if not.
         """
-        resp = self._cb.get_object(self._build_api_request_uri())
-        resp['cve_id'] = self._model_unique_id
-        self._info = resp
+        query = self._cb.select(Vulnerability).add_criteria("cve_id", self._model_unique_id)
+        results = list(query)
+
+        if len(results) > 1:
+            if self._info.get('os_product_id', None):
+                for vuln in results:
+                    if vuln.os_product_id == self._info["os_product_id"]:
+                        self._info = vuln._info
```

Comment:
This loop here was marked as "untested" in Coveralls. Maybe you could slip in another test for this? I saw that test coverage on this file had gone _down._ It was still up above 94%, but....

Reply:
Done